source/common/version/BUILD

```diff
@@ -18,6 +18,21 @@ genrule(
     visibility = ["//visibility:private"],
 )
 
+# This genrule generates a header file linking GIT_VERSION_NUMBER to version.h for labeling built binaries using the workspace status command output.
```

Comment:
Not sure if this is too long. See [the linter](https://github.com/envoyproxy/envoy/blob/main/support/README.md#fixing-format-problems).

Reply:
Ran linter locally 👍